File: webstore/webstore-java/src/main/java/com/renan/webstore/model/Category.java
package com.renan.webstore.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Entity
@Table(name = "CATEGORY")
public class Category {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "ID")
    private Integer id;
    @Column(name = "NAME", nullable = false)
    private String name;
    @ManyToOne
    @JoinColumn(name = "ID_UPPER_CATEGORY")
    private Category upCategory;
    
}

File: webstore/webstore-kotlin/src/main/kotlin/com/renan/webstore/model/Category.kt
package com.renan.webstore.model

import jakarta.persistence.*

@Entity
@Table(name = "CATEGORY")
data class Category(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "ID")
    val id: Int? = null,
    @Column(name = "NAME", nullable = false)
    var name: String,
    @ManyToOne
    @JoinColumn(name = "ID_UPPER_CATEGORY")
    var upperCategory: Category?
)
